File: src/leetcode/2026/06/Solution1047.java
public class Solution1047 {
    public String removeDuplicates(String s) {
        StringBuilder sb = new StringBuilder();
        for (char c : s.toCharArray()) {
            int length = sb.length();
            if (length > 0 && sb.charAt(length - 1) == c) {
                sb.deleteCharAt(length - 1);
            } else {
                sb.append(c);
            }
        }
        return sb.toString();
    }

    public static void main(String[] args) {
        Solution1047 solution = new Solution1047();

        String s1 = "abbaca";
        String result1 = solution.removeDuplicates(s1);
        System.out.println(result1); // Output: "ca"

        String s2 = "azxxzy";
        String result2 = solution.removeDuplicates(s2);
        System.out.println(result2); // Output: "ay"
    }
}

File: src/leetcode/2026/06/Solution1047.cpp
#include <string>

using namespace std;

class Solution1047 {
public:
    string removeDuplicates(string s) {
        string result;
        for (char c : s) {
            if (!result.empty() && result.back() == c) {
                result.pop_back();
            } else {
                result.push_back(c);
            }
        }
        return result;
    }
};

int main() {
    Solution1047 solution;

    string s1 = "abbaca";
    string result1 = solution.removeDuplicates(s1);
    // Output: "ca"
    printf("Input: %s, Output: %s\n", s1.c_str(), result1.c_str());

    string s2 = "azxxzy";
    string result2 = solution.removeDuplicates(s2);
    // Output: "ay"
    printf("Input: %s, Output: %s\n", s2.c_str(), result2.c_str());

    return 0;
}